indigo_api/models/works.py

```diff
@@ -407,6 +407,13 @@ def create_expression_at(self, user, date, language=None):
         doc.created_by_user = user
         doc.save()
 
+        # copy over attachments to new expression
+        attachments = Attachment.objects.filter(document=template)
```

Comment:
Are you sure this doesn't *move* the existing attachments?

Reply:
No, according to the docs, when you set the `pk` to `None` a new one will be created maintaining all previous details of the model.

Ref: https://docs.djangoproject.com/en/1.11/topics/db/queries/#copying-model-instances

Reply:
Ah, clever.

Hrm, I just tested it, and it doesn't quite work as expected because the file itself (`attachment.file`) doesn't get copied. In my local database, the attachment filename should change:

![indigo_api_attachment – indigo – localhost (1) 2020-04-06 18-41-13](https://user-images.githubusercontent.com/4178542/78582859-3d494300-7836-11ea-81b8-f6db0108a3c2.png)

158 is the old document id, 975 is the new one. So I think you'll need to actually create new attachment objects and/or copy the one file to the other. I always have to lookup how to do that with Django's FileField fields, which give you FieldFile objects.

Reply:
Let me recreate the attachment objects entirely instead

Reply:
Managed to copy the files: 
![image](https://user-images.githubusercontent.com/8082197/78597278-e6ebfc80-7855-11ea-98e9-08cfa4384b8e.png)

Just a single extra line, will push tomorrow

Reply:
@longhotsummer  Accessing a FileField returns a FieldFile object that wraps around around the file, and this can be saved a new file on the new document